### ISSUE TITLE
Endless error spam in MeshInstance3D

### ISSUE THREAD

**Issue by JekSun97** [godotengine/godot#91827]
### Tested versions

Originating in Godot 4.3 dev6, this problem does not occur in Godot 4.3 dev5

### System information

Godot v4.3.dev6 - Windows 10.0.19045 - GLES3 (Compatibility) - Radeon RX 560 Series (Advanced Micro Devices, Inc.; 31.0.14001.45012) - Intel(R) Core(TM) i5-4570 CPU @ 3.20GHz (4 Threads)

### Issue description

When selecting a mesh in MeshInstance3D, endless error spam occurs.
This only happens in compatibility mode.

https://github.com/godotengine/godot/assets/130399274/ebbce877-a3db-4411-9846-735bd1eb5681

### Steps to reproduce

1. Create a MeshInstance3D
2. Create a new mesh for him
3. Select this mesh

### Minimal reproduction project (MRP)

Easy to check

**Comment by pineapplemachine** [comment]
The error show in the video capture reads (repeatedly):

```
servers/rendering/storage/environment_storage.cpp:787 - Parameter "env" is null.
servers/rendering/storage/environment_storage.cpp:817 - Parameter "env" is null.
```

**Comment by JekSun97** [comment]
> The error show in the video capture reads (repeatedly):
> 
> ```
> servers/rendering/storage/environment_storage.cpp:787 - Parameter "env" is null.
> servers/rendering/storage/environment_storage.cpp:817 - Parameter "env" is null.
> ```

Are you implying that the scene lacks WorldEnvironment?
I have now created it with an Environment resource, but the error still occurs.

**Comment by akien-mga** [comment]
Thanks for the report. This seems to be a duplicate of #91239 which was fixed after 4.3-dev6, the fix will be in 4.3-beta1.

If you want, you can test latest GitHub Actions artifacts from the `master` branch to confirm that the bug is indeed fixed in your reproduction case.
